src/api/api.cpp

```diff
@@ -655,6 +662,47 @@ struct migraphx_experimental_custom_op
         }
         return (&out)->object;
     }
+
+    migraphx_experimental_custom_op_output_alias output_alias_f = nullptr;
+    std::ptrdiff_t output_alias(std::vector<migraphx::shape> inputs) const
+    {
+        std::remove_pointer_t<int64_t*> out;
+        if(output_alias_f == nullptr)
+            throw std::runtime_error("output_alias function is missing.");
+        std::array<char, 256> exception_msg;
+        exception_msg.front() = '\0';
+        auto api_error_result = output_alias_f(&out,
+                                               object_ptr.data,
+                                               exception_msg.data(),
+                                               exception_msg.size(),
+                                               object_cast<migraphx_shapes_t>(&(inputs)));
+        if(api_error_result != migraphx_status_success)
+        {
+            const std::string exception_str(exception_msg.data());
+            throw std::runtime_error("Error in output_alias of: " +
+                                     std::string(object_ptr.obj_typename) + ": " + exception_str);
+        }
+        return std::ptrdiff_t(out);
+    }
+
+    migraphx_experimental_custom_op_runs_on_offload_target runs_on_offload_target_f = nullptr;
+    bool runs_on_offload_target() const
+    {
+        std::remove_pointer_t<bool*> out;
+        if(runs_on_offload_target_f == nullptr)
+            throw std::runtime_error("runs_on_offload_target function is missing.");
+        std::array<char, 256> exception_msg;
+        exception_msg.front() = '\0';
+        auto api_error_result = runs_on_offload_target_f(
+            &out, object_ptr.data, exception_msg.data(), exception_msg.size());
```

Comment:
Small question, is the intent to use remove_pointer<bool *> since we want to reference the local context out as part of the offloaded call's result in runs_on_offload_target_f( &out, ....) and send it back out on the return?

Reply:
It is calling into C API. We pass everything by pointers so that if is compiled with a different C/C++ runtime library then,  it would have ABI compatibility.  This code is also auto-generated  `remove_pointer<bool*>` here is just an artifact of common logic for auto-gen code logic.

Reply:
Okay makes sense.